internal/kuberesource/resourcegen/main.go

```diff
@@ -68,7 +68,24 @@ func main() {
 		case "vault":
 			subResources = kuberesource.PatchRuntimeHandlers(kuberesource.Vault(*namespace), "contrast-cc")
 		case "gpu":
-			subResources = kuberesource.PatchRuntimeHandlers(kuberesource.GPU(), "contrast-cc")
+			if *rawPlatform == "" {
+				log.Fatalf("--platform must be set to one of %v", platforms.AllStrings())
+			}
+			var platform platforms.Platform
+			platform, err = platforms.FromString(*rawPlatform)
+			if err != nil {
+				log.Fatalf("Error parsing platform: %v", err)
+			}
+			var deviceURI string
+			switch platform {
+			case platforms.MetalQEMUTDXGPU:
+				deviceURI = "nvidia.com/GB100_B200"
+			case platforms.MetalQEMUSNPGPU:
+				deviceURI = "nvidia.com/GH100_H100_PCIE"
+			default:
+				log.Fatalf("platform %s does not support GPU tests", platform)
```

Comment:
This should probabkly move from here and the GPU test to `internal/platforms/platforms.go` or into `e2e/internal`